src/models/workout.js

```diff
@@ -125,27 +125,35 @@ const WorkoutModel = {
     }
   },
 
-  updateSetItems: async ({ workoutItemIdArray, workoutSetId }) => {
+  updateSetItems: async ({ workoutItemArray, userId }) => {
     try {
+      let { data: daysIdSet } = await supabase.from('workout_sets').select().match({ users_id: userId });
+
       let queryValForNew = [];
-      workoutItemIdArray.map(async (item) => {
-        if (item.id == null) {
-          let val = {
-            workout_items_id: item.workoutItemId,
-            workout_sets_id: Number(workoutSetId),
-            set_order: item.order,
-            reps: item.reps,
-            sets: item.sets,
-          };
-          queryValForNew.push(val);
-        } else {
-          let { error } = await supabase
-            .from('workout_set_items')
-            .update({ set_order: item.order, reps: item.reps, sets: item.sets })
-            .match({ id: item.id });
-
-          supabaseErrorCheck(error);
-        }
+      workoutItemArray.map(async (dayItems, i) => {
```

Comment:
change i and j to something better.

Reply:
```suggestion
      workoutItemArray.map(async (workoutItem, i) => {
```

Usually when you loop through something, it becomes singular of the original data you are looping